src/syscheckd/fim_db.c

```diff
@@ -714,7 +721,7 @@ int fim_db_data_checksum_range(fdb_t *fim_sql, const char *start, const char *to
             goto end;
         }
         entry = fim_db_decode_full_row(fim_sql->stmt[FIMDB_STMT_GET_PATH_RANGE]);
-        if (i == (m - 1)) {
+        if (i == (m - 1) && entry->path) {
```

Comment:
In the function `fim_db_decode_full_row()` please, change the use of `w_strdup()` to `os_strdup()`.